src/inpar/4C_inpar_scatra.hpp

```diff
@@ -259,6 +260,11 @@ namespace Inpar
     /// set additional scatra conditions
     void set_valid_conditions(std::vector<Core::Conditions::ConditionDefinition>& condlist);
 
+    std::string impltype_to_string(ImplType impltype);
+
+    Core::IO::InputSpec get_impl_type_input_spec(const std::string& name,
+        Core::IO::InputSpecBuilders::ScalarData<ImplType> data,
```

Comment:
Hm, I was treating `ScalarData` as an implementation detail. It's not internal, since it is obviously needed, although you typically don't name it explicitly. You only use this function in one place. How about just creating the map there? Or is this a recurring building block, you think? Then a utility to create such a map could also be part of the `InputSpec` framework.